doc/source/usage.rst

```diff
@@ -0,0 +1,194 @@
+.. _usage:
+
+Usage
+=====
+
+Overview
+--------
+
+This sections dives deeper into the inner workings of **DataStructureDefinition** and
+**RegionProcessor** as well as their associated auxiliary classes.
+
+
+Yaml file schemas
+--------------------
+
+Both classes read information from yaml files which are required to follow specific
+structures.
+
+For **DataStructureDefinition** there are three types of schemas: variable, region and tag.
+
+Variable
+~~~~~~~~
+
+The *variable* codelist of the **DataStructureDefinition** will be read from all yaml
+files located in a folder of that name (including any sub-folders). They must be
+formatted as a list of dictionaries mapping the variable (key) to its attributes.
+
+.. code:: yaml
+
+   - Some Variable:
+      unit: A unit
+      description: A short description
+      <other attribute>: Some text (optional)
+
+**Notes**
+
+* Every variable **must have** a **unit**, which should be compatible with the
+  Python package `iam-units <https://github.com/iamconsortium/units>`_.
+
+* The unit attribute can also be empty, i.e., the variable is *dimensionless*.
+
+* All other attributes such as "description" are optional. It is recommended to provide
+  a description for better documentation.
+
+* For region aggregation, the pyam function `aggregate_region
+  <https://pyam-iamc.readthedocs.io/en/stable/api/iamdataframe.html#pyam.IamDataFrame.aggregate_region>`_
+  is used. Attributes provided in a mapping that match the parameters "components",
+  "method", "weight" or "drop_negative_weights" will be used in the aggregation.
+
+Region
+~~~~~~
+
+To avoid repeating a “hierarchy” attribute many times (e.g., country,
+continent), the yaml files must have a nested dictionary structure:
+
+.. code:: yaml
+
+   - <Hierarchy Level>:
+     - region 1:
+         Attribute: Attribute value
+     - region 2:
+         Attribute: Attribute value 
+
+**Notes**
+
+* Every region **must be** defined as part of a hierarchy. 
+
+* When importing the codelist, the hierarchy will be added as attribute,
+  such that it can be retrieved as:
+
+.. code:: python
+
+   DataStructureDefinition.region["Region Name"]["Hierarchy"] = "<Hierarchy Level>"
+
+* Other attributes specified in the yaml file can include (for countries)
+  ISO2/3-codes, or the list of countries included in a macro-region (i.e.,
+  a continent or large region).
+
```

Comment:
"Generic" format should also be mentioned. This can be used, for example, to specify codelists for the allowed scenario names.

Reply:
Added generic to the list